config/config.go

```diff
@@ -32,6 +33,57 @@ type ServerConfig struct {
 	KubeConfigs     map[string]*api.Config     `json:"kubeConfigs"`
 }
 
+// LoadFromPath attempts to load a config from the given path. If the config
+// doesn't exist, an empty config is returned.
+func LoadFromPath(path string) (Config, error) {
+	cf := Config{
+		Path:    path,
+		Servers: make(map[string]*ServerConfig),
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		// it's okay if the file is empty, we still return a valid config
+		if os.IsNotExist(err) {
+			return cf, nil
+		}
+
+		return cf, err
+	}
+
+	if err := json.Unmarshal(content, &cf); err != nil {
+		return cf, fmt.Errorf("unmarshaling %s: %w", path, err)
+	}
+	cf.Path = path
```

Comment:
I don't think this is needed, since you do it above already.

Reply:
The only concern here is whether `path` was ever written to the config file. Looking at the git history, it seems like it was always `omitempty` so we can probably assume that `path` was never written. In that case, it might be okay to remove that line.. In fact, I think `Config` should be split into 2 separate struct, one with **only** valid config file fields and the other with more stuff like path. The question is how much change do we want to make here? Should we put more time into this config module now, or later?

@MbolotSuse Can you confirm when we want this in? Do we want it for bug complete 2.8 (which is today) or should we take more time to flesh this out a bit more?

Reply:
I'm thinking something like:

```go
// Already hardcoded currently in cmd/common.go
const cfgFile = "cli2.json"
type Environment struct {
	dir string
}

func NewEnvironment() *Environment {}
// LoadConfig gets the config (<dir>/cli2.json) or creates it.
func (e *Environment) LoadConfig() (Config, error) {} 
// SaveConfig saves the config to <dir>/cli2.json
func (e *Environment) SaveConfig(Config) error {}

type Config struct {
	Servers map[string]*ServerConfig
	CurrentServer string
}
```

so 
- Config no longer has `Path` which wasn't written to the config file anyway
- We can move cfgFile from `cmd/common.go` to `config/config.go`. It's fine to be hardcoded because the user can't change it anyway (the `--config` flag specifies the directory, not the file)
- We can remove those hacky Path reset stuff in the tests

Reply:
The change that you are talking about here seems more significant. I would say that we should try and keep the blast radius on this change small.

Reply:
Agreed.